### PR TITLE
Collect historical data

### LINKED ISSUE #2: Create a script for a virtual Python env to run benchmark tasks
Subtask of #1 

The idea is to create a script that will be run one time to gather historical data for average of circuit depth and gate count, using experiment code documented [here](https://github.com/qiskit-community/submit-metriq/issues/1#issuecomment-1507586579).
Once the backfill is complete, this structure can be used to run any benchmark tasks on newer qiskit versions. This automation on future qiskit releases will be implemented as a separate PR.

- Get all qiskit versions from pip command `pip index versions qiskit `and filter those since 2020-03-24 (last update in metriq), hard code it for one time run to build historical data.
- The entry point is a script that creates on tox environment per qiskit version. Input parameter is qiskit version (later on, also python version and the benchmark script - since in the future there can be other tasks).
- The script creates a tox environment with the given qiskit version, loads the benchmark script, run it N times (starting at 100) and outputs the average of circuit depth and gate count (first print to screen, later it can output to a file).
- Once the job is complete, the environment is shut down and cleaned.

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Closes #2 

This PR:
- [x] Implements benchmark task 25 as documented [here](https://github.com/qiskit-community/submit-metriq/issues/1#issuecomment-1507586579)
- [x] Retrieves qiskit-terra releases info from https://pypi.org/pypi/qiskit-terra/json for version number, date and python version
- [x] Filters versions since 2020-03-24 (last update in metriq), excluding RCs and patch releases
For every filtered qiskit-terra version:
- [x] creates a tox environment with respect python version
- [x] runs benchmark task  100 times and get average results of gate count and circuit depth

**Requirements:**
- python3
- tox

**Run main script:**
`python scripts/run_tox_envs.py`

### CODE REVIEW

File: scripts/run_tox_envs.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+for info in versions_info:
+    version = info["version"]
+    env_name = "qiskit v" + version
+
+    # TODO: This does not override latest version defined in tox.ini. Keep looking into this.
+    command = "tox -e "+ env_name + " -- pip install --qiskit-terra==" + version
+    # CLI commands tried:
+    # tox -e q_0.20.0 -- pip install qiskit-terra==0.20.0
+    # tox -e q_0.20.0 --force-dep qiskit-terra==0.20.0
+    # tox -e q_0.20.0 -r --force-dep qiskit==0.20.0
```
Comment (karlaspuldaro) [inline scripts/run_tox_envs.py]: @1ucian0 Any ideas how to override the latest qiskit version when creating a tox env using CLI?

File: scripts/run_tox_envs.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+usedevelop = True
+
+deps =
+qiskit-terra=={}
+requests
+pyzx
+
+commands = 
+    python {}/scripts/circuit_depth_and_gate_count.py
+""".format(env_name, qiskit_version, "{toxinidir}")
```
Comment (1ucian0) [inline scripts/run_tox_envs.py]: Consider using f-strings. Something like this:
```suggestion
    tox_config = f"""
[tox]
minversion = 3.7
envlist = {env_name}

[testenv]
usedevelop = True

deps =
qiskit-terra=={qiskit_version}
requests
pyzx

commands = 
    python {toxinidir}/scripts/circuit_depth_and_gate_count.py
"""
```

File: scripts/circuit_depth_and_gate_count.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    dag = self._do_pass(pass_, dag, passset.options)
+  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/runningpassmanager.py", line 173, in _do_pass
+    dag = self._run_this_pass(pass_, dag)
+  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/runningpassmanager.py", line 202, in _run_this_pass
+    new_dag = pass_.run(dag)
+  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passes/utils/gate_direction.py", line 299, in run
+    return self._run_coupling_map(dag, layout_map)
+  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passes/utils/gate_direction.py", line 177, in _run_coupling_map
+    f"Flipping of gate direction is only supported "
+qiskit.transpiler.exceptions.TranspilerError: "Flipping of gate direction is only supported for ['cx', 'cz', 'ecr'] at this time, not 'swap'."
```
Comment (karlaspuldaro) [inline scripts/circuit_depth_and_gate_count.py:108]: @1ucian0 running this task in older qiskit-terra versions results in this transpile error - this was using v0.23.0. I want to double check in other older versions as well if the same error persists.

File: scripts/run_tox_envs.py
Comment (1ucian0) [inline scripts/run_tox_envs.py:1]: I never use it, but tox has an API to launch envs from Python, without using `os.system`: https://tox.wiki/en/latest/plugins_api.html

**Pull request merged**